internal/model/rwfolder.go

```diff
@@ -1239,39 +1256,63 @@ func (p *rwFolder) dbUpdaterRoutine() {
 		maxBatchTime = 2 * time.Second
 	)
 
-	batch := make([]protocol.FileInfo, 0, maxBatchSize)
+	batch := make([]dbUpdateJob, 0, maxBatchSize)
+	files := make([]protocol.FileInfo, 0, maxBatchSize)
 	tick := time.NewTicker(maxBatchTime)
 	defer tick.Stop()
 
+	handleBatch := func() {
+		found := false
+		var lastFile protocol.FileInfo
+
+		for _, job := range batch {
+			files = append(files, job.file)
+			if job.file.IsInvalid() || (job.file.IsDirectory() && !job.file.IsSymlink()) {
+				continue
+			}
+
+			if job.jobType&(dbUpdateHandleFile|dbUpdateDeleteFile) == 0 {
+				continue
+			}
+
+			found = true
+			lastFile = job.file
+		}
```

Comment:
This might be a premature optimization, but I'd probably do the loop backwards here as the odds are probably larger that a given FileInfo is the one want than that it isn't... I.e. we should be able to stop the loop quicker that way I think.